cmd/kube-scheduler/app/server_test.go

```diff
@@ -0,0 +1,352 @@
+			{Name: "AzureDiskLimits"},
+			{Name: "VolumeBinding"},
+			{Name: "VolumeZone"},
+			{Name: "InterPodAffinity"},
+			{Name: "PodTopologySpread"},
+		},
+		"PreScorePlugin": {
+			{Name: "InterPodAffinity"},
+			{Name: "DefaultPodTopologySpread"},
+			{Name: "TaintToleration"},
+			{Name: "PodTopologySpread"},
+		},
+		"ScorePlugin": {
+			{Name: "NodeResourcesBalancedAllocation", Weight: 1},
+			{Name: "ImageLocality", Weight: 1},
+			{Name: "InterPodAffinity", Weight: 1},
+			{Name: "NodeResourcesLeastAllocated", Weight: 1},
+			{Name: "NodeAffinity", Weight: 1},
+			{Name: "NodePreferAvoidPods", Weight: 10000},
+			{Name: "DefaultPodTopologySpread", Weight: 1},
+			{Name: "TaintToleration", Weight: 1},
+			{Name: "PodTopologySpread", Weight: 1},
+		},
+		"BindPlugin": {{Name: "DefaultBinder"}},
+	}
+
+	testcases := []struct {
+		name         string
+		flags        []string
+		wantProfiles []schedulerProfile
+	}{
+		{
+			name: "config file",
+			flags: []string{
+				"--config", configFile,
+				"--kubeconfig", configKubeconfig,
+			},
+			wantProfiles: []schedulerProfile{
+				{
+					SchedulerName: v1.DefaultSchedulerName,
+					Plugins:       defaultPlugins,
+				},
+			},
+		},
+		{
+			name: "v1alpha1 config with SchedulerName and HardPodAffinitySymmetricWeight",
+			flags: []string{
+				"--config", v1alpha1Config,
+				"--kubeconfig", configKubeconfig,
+			},
+			wantProfiles: []schedulerProfile{
+				{
+					SchedulerName: schedulerName,
+					Plugins:       defaultPlugins,
+				},
+			},
+		},
+		{
+			name: "plugin config with single profile",
+			flags: []string{
+				"--config", pluginConfigFile,
+				"--kubeconfig", configKubeconfig,
+			},
+			wantProfiles: []schedulerProfile{
+				{
+					SchedulerName: v1.DefaultSchedulerName,
+					Plugins: map[string][]kubeschedulerconfig.Plugin{
+						"BindPlugin":      {{Name: "DefaultBinder"}},
+						"FilterPlugin":    {{Name: "NodeResourcesFit"}, {Name: "NodePorts"}},
+						"PreFilterPlugin": {{Name: "NodeResourcesFit"}, {Name: "NodePorts"}},
+						"PreScorePlugin":  {{Name: "InterPodAffinity"}, {Name: "TaintToleration"}},
+						"QueueSortPlugin": {{Name: "PrioritySort"}},
+						"ScorePlugin":     {{Name: "InterPodAffinity", Weight: 1}, {Name: "TaintToleration", Weight: 1}},
+					},
+				},
+			},
+		},
+		{
+			name: "plugin config with multiple profiles",
+			flags: []string{
+				"--config", multiProfilesConfig,
+				"--kubeconfig", configKubeconfig,
+			},
+			wantProfiles: []schedulerProfile{
+				{
+					SchedulerName: profileName1,
+					Plugins:       defaultPlugins,
+				},
+				{
+					SchedulerName: profileName2,
+					Plugins: map[string][]kubeschedulerconfig.Plugin{
+						"BindPlugin":      {{Name: "DefaultBinder"}},
+						"QueueSortPlugin": {{Name: "PrioritySort"}},
+					},
+				},
+			},
+		},
+		{
+			name: "Deprecated SchedulerName flag",
+			flags: []string{
+				"--kubeconfig", configKubeconfig,
+				"--scheduler-name", schedulerName,
```

Comment:
Are there any non-deprecated flags that could be easy to verify?

Reply:
added test for `write-to-config` flag

Reply:
we can also add a test for the feature flag, but can we do that in a follow up PR? I find it is a little tricky to restore feature flags for different test cases.